ingester2/src/server/grpc/query.rs

```diff
@@ -194,7 +201,8 @@ where
             }
         };
 
-        let output = encode_response(response, self.ingester_id).map_err(tonic::Status::from);
+        let output =
+            encode_response(response, self.ingester_id, encoding_span).map_err(tonic::Status::from);
```

Comment:
Tiny simplification:

```suggestion
        let output =
            encode_response(response, self.ingester_id, span).map_err(tonic::Status::from);
```

And add `.clone()` on the span where it's used above (line 189) instead of at the intro to this func - it's clearer that it's all the same span that way! Less vars == less thinking.